src/native/corehost/comhost/comhost.cpp

```diff
@@ -135,7 +221,15 @@ COM_API HRESULT STDMETHODCALLTYPE DllGetClassObject(
         iter->second.type.c_str(),
         (void**)&classFactory
     };
-    RETURN_IF_FAILED(act(&cxt));
+    if (del_has_load_context)
+    {
+        RETURN_IF_FAILED(act(&cxt, load_context));
```

Comment:
Here I would take the `void * delegate` we got from the `get_com_delegate`, cast it to the right function pointer and then call. Just like we do in the else branch.